client/src/intro-exit/CheckUnderstanding.jsx

```diff
@@ -19,207 +20,230 @@ export function Radio({ selected, name, value, label, onChange }) {
-      setTask(e.target.value);
-    }
+  function handleTime(e) {
+    setTime(e.target.value);
+  }
 
-    function handleTasksTwo(e) {
-      setTaskTwo(e.target.value);
-    }
+  function handleTasks(e) {
+    setTask(e.target.value);
+  }
 
-    function handleResponse(e) {
-      setResponse(e.target.value);
-    }
+  function handleTasksTwo(e) {
+    setTaskTwo(e.target.value);
+  }
 
-    function handleVideo(e) {
-        setVideo(e.target.value);
-    }
+  function handleResponse(e) {
+    setResponse(e.target.value);
+  }
 
-    function handleSubmit(event) {
-      if (allowContinue) {
-        console.log("Intro Quiz submitted correctly")
-        next();
-      } else {
-        setShowIntro(true);
-        console.log("Intro Quiz submitted with errors")
-        alert("Some of your answers were incorrect")
-      }
-      event.preventDefault();
+  function handleVideo(e) {
+      setVideo(e.target.value);
+  }
+
+  function handleSubmit(event) {
+    setIncorrectResponses(false)
+    if (allowContinue) {
+      console.log("Understanding check submitted correctly")
+      next();
+    } else {
+      console.log("Understanding check submitted with errors, throwing alert")
+      setIncorrectResponses(true)
+      document.getElementById("container").scrollIntoView(true)
+      window.scrollTo(0, 0)
     }
-    
+    event.preventDefault()
+  }
 
-    return (
-      <div className="ml-5 mt-1 sm:mt-5 p-5 basis-1/2">
-        <form className="space-y-8 divide-y divide-gray-200" onSubmit={handleSubmit}>
-            <div>
-                <h3 className="text-lg leading-6 font-medium text-gray-900">
-                   Please verify that you understand the instructions:
-                </h3>
-            </div>
-            <div>
-              <label className="block text-md font-medium text-gray-700 my-2">
-                What will you do in this task?
-              </label>
-              <div className="ml-5 grid gap-2">
-                <Radio
-                  selected={task}
-                  name="task"
-                  value="dishwasher"
-                  label="Eat a bagel"
-                  onChange={handleTasks}
-                />
-                <Radio
-                  selected={task}
-                  name="task"
-                  value="transcribe"
-                  label="Transcribe a discussion"
-                  onChange={handleTasks}
-                />
-                <Radio
-                  selected={task}
-                  name="task"
-                  value="correct"
-                  label="Discuss a topic with others"
-                  onChange={handleTasks}
-                />
-                <Radio
-                  selected={task}
-                  name="task"
-                  value="read"
-                  label="Proofread a paper about group discussions"
-                  onChange={handleTasks}
-                />
+  class CheckUnderstandingForm extends React.Component {
```

Comment:
I think we can do this using the normal pattern, without having to build up this class. The syntax should be something like 
```jsx
{incorrectResponses && <Alert ...>
```
If the `incorrectResponses` is false, it short-circuits the evaluation of the rest of the line, and so doesn't display the component. A good example is here: https://github.com/Watts-Lab/deliberation-empirica/blob/main/client/src/intro-exit/VideoCheck.jsx#L111

There are some interesting differences between using classes and purely functional react components, which I've been learning about this summer (in empirica V1, the react components are all classes) but I think I'd rather stick with purely functional components for now. =)